experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -1,3 +1,189 @@
+@doc raw"""
+    exchange!(W::WeylGroup, i::UInt8, w::AbstractVector{UInt8}) -> Bool
+
+Return `true`, if `w` could be transformed into an equivalent word with `i` at the first position,
+otherwise `false`.
+
+See also [`exchange!(::WeylGroup, ::Vector{UInt8}, ::UInt8)`](@ref).
+"""
+function exchange!(W::WeylGroup, i::UInt8, w::AbstractVector{UInt8})
+  root = i
+  for s in 1:length(w)
+    if w[s] == root
+      w[2:s] = w[1:(s - 1)]
+      w[1] = i
+      return true
+    end
+    root = W.refl[Int(w[s]), Int(root)]
+  end
+
+  return false
+end
+
+@doc raw"""
+    exchange!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8) -> Bool
+
+Return `true`, if `w` could be transformed into an equivalent word with `i` at the last position,
+otherwise `false`.
+
+See also [`exchange!(::WeylGroup, ::UInt8, ::Vector{UInt8})`](@ref).
+"""
+function exchange!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8)
+  root = i
+  for s in length(w):-1:1
+    if w[s] == root
+      w[s:(end - 1)] = w[(s + 1):end]
+      w[end] = i
+      return true
+    end
+    root = W.refl[Int(w[s]), Int(root)]
+  end
+
+  return false
+end
+
+@doc raw"""
+    braid_moves(W::WeylGroup, w1::Vector{UInt8}, w2::Vector{UInt8}) -> Vector{Tuple{Int,Int,Int}}
+
+Return the braid moves required to transform the string `j` into `i` with respect to the Weyl group `W`.
+A braid move `(n, len, dir)` should be understood as follows:
+- `n` is the position where the braid move starts
+- `len` is the length of the braid move
+- `dir` is the direction of the braid move. If `len=2` or `len=3`, `dir` is `0` or `-1`.
+  If `len=4` or `len=6`, `dir` is `-2` or `-3` if the root at `n` is short, otherwise `dir` is `-1`.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:A, 3);
+
+julia> braid_moves(W, UInt8[1,2,1,3,2,1], UInt8[1,3,2,1,3,2])
+4-element Vector{Tuple{Int64, Int64, Int64}}:
+ (4, 2, 0)
+ (2, 3, -1)
+ (4, 3, -1)
+ (3, 2, 0)
+
+julia> W = weyl_group(:B, 2);
+
+julia> braid_moves(W, UInt8[2,1,2,1], UInt8[1,2,1,2])
+1-element Vector{Tuple{Int64, Int64, Int64}}:
+ (1, 4, -2)
+```
+"""
+function braid_moves(W::WeylGroup, w1::Vector{UInt8}, w2::Vector{UInt8})
+  return _braid_moves(W, w1, w2, 0)
```

Comment:
```suggestion
  return _braid_moves(W, w1, w2, 0)::Vector{Tuple{Int64, Int64, Int64}}
```
To help inference, since `_braid_moves` may also return `nothing`, but `braid_moves` should instead return an empty vector